trin-core/src/types/receipts.rs

```diff
@@ -32,6 +33,62 @@ impl Receipts {
     }
 }
 
+impl ssz::Encode for Receipts {
+    fn is_ssz_fixed_len() -> bool {
+        false
+    }
+
+    fn ssz_append(&self, buf: &mut Vec<u8>) {
+        ReceiptsHelper::from(self).ssz_append(buf);
+    }
+
+    fn ssz_bytes_len(&self) -> usize {
+        self.as_ssz_bytes().len()
+    }
+}
+
+impl ssz::Decode for Receipts {
+    fn is_ssz_fixed_len() -> bool {
+        false
+    }
+
+    fn from_ssz_bytes(bytes: &[u8]) -> Result<Self, ssz::DecodeError> {
+        let helper = ReceiptsHelper::from_ssz_bytes(bytes)?.try_into()?;
+        Ok(helper)
+    }
+}
+
+#[derive(Debug, Decode, Encode)]
+struct ReceiptsHelper {
+    rlp_receipt_list: Vec<Vec<u8>>,
+}
+
+impl From<&Receipts> for ReceiptsHelper {
+    fn from(receipts: &Receipts) -> Self {
+        let rlp_receipts: Vec<Vec<u8>> = receipts
+            .receipt_list
+            .iter()
+            .map(|receipt| rlp::encode(receipt).to_vec())
+            .collect();
+        Self {
+            rlp_receipt_list: rlp_receipts,
+        }
+    }
+}
+
+impl TryFrom<ReceiptsHelper> for Receipts {
+    type Error = ssz::DecodeError;
+
+    fn try_from(helper: ReceiptsHelper) -> Result<Self, Self::Error> {
+        let receipt_list: Vec<Receipt> = helper
+            .rlp_receipt_list
+            .iter()
+            .map(|bytes| rlp::decode(bytes).unwrap())
```

Comment:
To my eyes, it's very strange to me that this works. The typed receipts are specifically built so that they cannot be rlp-decoded. I guess it's because there is this magic inside of the `Receipt` class that clips the type byte out before rlp-decoding.

At some point there could be receipts that don't have any RLP encoding in them at all. Following this pattern we've started would still call it `rlp::decode()` here anyway, which feels wrong. So something like `Receipt::decode(bytes)` here would be more satisfying to my eyes.